### ISSUE TITLE
We need a single method for listing ready nodes which will be invoked by plugins

### ISSUE THREAD

**Issue by knelasevero** [kubernetes-sigs/descheduler#941]
`Nodes` param is a tricky param. It can not be passed through the plugin arguments since a plugin is initialized before its extension point is invoked. In my PoC implementation I reached for the current nodes through the shared informer: https://github.com/kubernetes-sigs/descheduler/pull/781/files#diff-66db71b98f61a3c06ca8d17cbed99626fc23f8bab081cd71a8baf0867613d4d5R111. I am open to suggestions on how to make it better. Using the shared node informer directly is acceptable solution for the moment. Nevertheless, once the descheduler is migrated into the framework completely, we need a single method for listing ready nodes which will be invoked by this plugin and the main descheduling loop. And possibly by others who needs to have identical list of ready nodes.

_Originally posted by @ingvagabund in https://github.com/kubernetes-sigs/descheduler/pull/929#discussion_r962619304_

**Comment by damemi** [comment]
Based on the comment in https://github.com/kubernetes-sigs/descheduler/pull/781/files#diff-66db71b98f61a3c06ca8d17cbed99626fc23f8bab081cd71a8baf0867613d4d5R111, it sounds like the big question around this is whether to cache the list of nodes before the strategy runs, or pull them live from within the strategy. Is that basically it?

**Comment by knelasevero** [comment]
Yeah, I think that's it. Not sure the title is good here :sweat_smile:  I created the issue quickly with the 'create issue from comment' github feature

**Comment by knelasevero** [comment]
lifecycle frozen
